openslides/core/views.py

```diff
@@ -46,6 +56,192 @@ def get(self, *args, **kwargs):
         return HttpResponse(content)
 
 
+class ProjectorViewSet(ReadOnlyModelViewSet):
+    """
+    API endpoint to list, retrieve and update the projector slide info.
+    """
+    queryset = Projector.objects.all()
+    serializer_class = ProjectorSerializer
+
+    def check_permissions(self, request):
+        """
+        Calls self.permission_denied() if the requesting user has not the
+        permission to see the projector and in case of an update request the
+        permission to manage the projector.
+        """
+        manage_methods = (
+            'activate_projector_elements',
+            'clear_and_activate_projector_elements',
+            'deactivate_projector_elements',
+            'deactivate_all_projector_elements')
+        if (not request.user.has_perm('core.can_see_projector') or
+                (self.action in manage_methods and
+                 not request.user.has_perm('core.can_manage_projector'))):
+            self.permission_denied(request)
+
+    @detail_route(methods=['post'])
+    def activate_projector_elements(self, request, pk):
+        """
+        REST API operation to activate projector elements. It expects a POST
+        request to /rest/core/projector/<pk>/activate_projector_elements/
+        with list of dictionaries to append to the projector config entry.
+        """
+        # Get config entry form projector model, add new elements and try to
+        # serialize. This raises ValidationErrors if the data is invalid.
+        projector_instance = self.get_object()
+        projector_config = projector_instance.config
+        for projector_element in request.data:
+            projector_config.append(projector_element)
+        serializer = self.get_serializer(projector_instance, data={'config': projector_config}, partial=False)
+        serializer.is_valid(raise_exception=True)
```

Comment:
I think this depends on the exception. If something like Http404 is raised, the user gets HTTP 404. I have to tested this but used the code as the framework does in `mixins.py`.